client/packages/system/src/Patient/Insurance/components/InsuranceModal/InsuranceModal.tsx

```diff
@@ -0,0 +1,222 @@
+  const formatDateTime = useFormatDateTime();
+  const { success, error } = useNotification();
+  const { current, setModal } = usePatientModalStore();
+
+  const { Modal } = useDialog({
+    disableBackdrop: true,
+    onClose: () => setModal(undefined),
+    isOpen: current === PatientModal.Insurance,
+  });
+
+  const nameId = usePatient.utils.id();
+  const { data } = usePatient.document.insurances({ nameId });
+  const [insurance, setInsurance] = useState(DEFAULT_INSURANCE);
+
+  const { urlQuery } = useUrlQuery();
+  const insuranceId = urlQuery['insuranceId'];
+  const haveInsuranceId = insuranceId !== undefined;
+  const selectedInsurance = data?.nodes.find(({ id }) => id === insuranceId);
+
+  const { mutateAsync: updateInsurance } =
+    usePatient.document.updateInsurance();
+
+  const handleInsuranceUpdate = async (): Promise<void> => {
+    const { policyNumberFamily, policyNumberPerson, ...insuranceInput } =
+      insurance;
+    try {
+      await updateInsurance(insuranceInput);
+      success(t('messages.insurance-saved'))();
+    } catch (e) {
+      error(
+        t('messages.error-saving-insurances', { error: (e as Error).message })
+      )();
+    }
+  };
+
+  const { mutateAsync: insertInsurance } =
+    usePatient.document.insertInsurance();
+
+  const handleInsuranceInsert = async (): Promise<void> => {
+    try {
+      await insertInsurance({
+        ...insurance,
+        id: FnUtils.generateUUID(),
+        nameId,
+      });
+      success(t('messages.insurance-created'))();
+    } catch (e) {
+      error(
+        t('messages.error-saving-insurances', { error: (e as Error).message })
+      )();
+    }
+  };
+
+  const handleSave = async (): Promise<void> => {
+    if (insuranceId !== undefined) await handleInsuranceUpdate();
+    else await handleInsuranceInsert();
+    setModal(undefined);
+  };
+
+  useEffect(() => {
+    if (selectedInsurance) {
+      const { insuranceProviders, policyNumber, ...insuranceDetails } =
+        selectedInsurance;
+
+      setInsurance({
+        ...insuranceDetails,
+        policyNumberFamily: insuranceDetails.policyNumberFamily ?? '',
+        policyNumberPerson: insuranceDetails.policyNumberPerson ?? '',
+      });
+    } else {
+      setInsurance(DEFAULT_INSURANCE);
+    }
+  }, [selectedInsurance]);
+
+  return (
+    <Modal
+      width={800}
+      title={
+        insuranceId === undefined
+          ? t('title.new-insurance')
+          : t('title.edit-insurance')
+      }
+      cancelButton={
+        <DialogButton variant="cancel" onClick={() => setModal(undefined)} />
+      }
+      okButton={<DialogButton variant="save" onClick={handleSave} />}
+      sx={{
+        '& .MuiDialogContent-root': { display: 'flex', alignItems: 'center' },
+      }}
+    >
+      <Stack gap={8} flexDirection="row">
+        <Box display="flex" flexDirection="column" gap={2}>
+          <InputWithLabelRow
+            label={t('label.policy-number-family')}
```

Comment:
Do these terms mean anything to an average user? I know we're mapping these fields to OG fields, but I can't help thinking we could capture the number somewhat nicer? Even in OG they don't explicitly name these fields in the user input:

![Screenshot 2025-02-19 at 12 10 39 PM](https://github.com/user-attachments/assets/0fe0b7b6-60f2-4a7a-b92e-3e7a5185c632)

Can probably make a new issue to address this if you want.

Reply:
Issue for the above: https://github.com/orgs/msupply-foundation/projects/21/views/7?pane=issue&itemId=98633150&issue=msupply-foundation%7Copen-msupply%7C6573